src/odemis/dataio/csv.py

```diff
@@ -132,6 +132,26 @@ def export(filename, data):
             csv_writer.writerow(headers)
             csv_writer.writerows(rows)
 
+    elif data.metadata.get(model.MD_ACQ_TYPE, None) == model.MD_AT_EK:
+        logging.debug("Exporting angular spectrum data to CSV")
+
+        spectrum_range, unit_c = spectrum.get_spectrum_range(data)
+        if unit_c == "m":
+            spectrum_range = [s * 1e9 for s in spectrum_range]
+            unit_c = "nm"
+
+        angle_range, _ = spectrum.get_angle_range(data)
+        unit_a = "°"
+        angles = [math.degrees(theta) for theta in angle_range if not math.isnan(theta)]  # Convert radians to degrees
```

Comment:
I don't think you can (and need to) drop the NaN here. Normally the NaN should have already be discarded. In addition, if you drop the angles but not the data, it's going to not be matching.